### ISSUE TITLE
Error: Unhandled Rejection

### ISSUE THREAD

**Issue by AchrafYndz** [reisxd/revanced-builder#27]
Error when running the `node .`  command.

Followed all steps as found in `README.MD`, jdk-17 is installed.

Here's the error:
![Screenshot error terminal](https://user-images.githubusercontent.com/92723683/177407769-a2497ec4-f413-4fa6-9ade-4084a75fd86e.png)

```bash
Unhandled Rejection
Reason: TypeError: Cannot read property 'includes' of undefined
Please make an issue at https://github.com/reisxd/revanced-builder/issues.
Press any key to exit...
```

**Comment by reisxd** [comment]
> Error when running the `node .`  command.
> 
> Followed all steps as found in `README.MD`, jdk-17 is installed.
> 
> Here's the error:
> ![Screenshot error terminal](https://user-images.githubusercontent.com/92723683/177407769-a2497ec4-f413-4fa6-9ade-4084a75fd86e.png)
> 
> 
> ```bash
> Unhandled Rejection
> Reason: TypeError: Cannot read property 'includes' of undefined
> Please make an issue at https://github.com/reisxd/revanced-builder/issues.
> Press any key to exit...
> ```

Hey, can you change the `unhandledRejection` event name to something else? So you can get the actual error.

**Comment by kzshantonu** [comment]
Can confirm I have the same error

**Comment by reisxd** [comment]
@jgmGit and @kzshantonu This error isn't an "error" but a log that an error occured.

**Comment by jgmGit** [comment]
@reisxd how can we see what went wrong ?

**Comment by reisxd** [comment]
> @reisxd how can we see what went wrong ?

You didn't post any console output, but for @jgmGit:
In v1.2.5, theres 3 times that the program access an arrays first element (array[0]) and I don't see anything wrong with the code. The first check checks if ADB and Zulu JDK 17 is installed and it obviously passed. Other one? It only checks it for getting the YT version for rooted builds. So, I have no idea.

**Comment by jgmGit** [comment]
Is there some way to enable more verbose logging ?

**Comment by reisxd** [comment]
> Is there some way to enable more verbose logging ?

You can run from source, modify the `unhandledRejection` event name to something else and itll show the actual error.

**Comment by jgmGit** [comment]
@reisxd  the first error i posted is from v1.2.5

```
./revanced-builder-nodejs-linux 
Downloading latest patches, integrations and cli...
[revanced-cli-2.3.2-all.jar] ✓
[revanced-patches-2.9.4.dex] ✓
[revanced-patches-2.9.4.jar] ✓
[app-release-unsigned.apk] ✓
[microg.apk] ✓
No device found! Fallback to only build.
Unhandled Rejection
Reason: TypeError: Cannot read properties of null (reading '0')
Please make an issue at https://github.com/reisxd/revanced-builder/issues.
Press any key to exit...
```

The second one I posted was from v1.2.4, there the microg gets installed, but the apk not.

**Comment by reisxd** [comment]
> 
> @reisxd  the first error i posted is from v1.2.5
> 
> ```
> ./revanced-builder-nodejs-linux 
> Downloading latest patches, integrations and cli...
> [revanced-cli-2.3.2-all.jar] ✓
> [revanced-patches-2.9.4.dex] ✓
> [revanced-patches-2.9.4.jar] ✓
> [app-release-unsigned.apk] ✓
> [microg.apk] ✓
> No device found! Fallback to only build.
> Unhandled Rejection
> Reason: TypeError: Cannot read properties of null (reading '0')
> Please make an issue at https://github.com/reisxd/revanced-builder/issues.
> Press any key to exit...
> ```
> 
> The second one I posted was from v1.2.4, there the microg gets installed, but the apk not.

Try doing what i said.

**Comment by jgmGit** [comment]
@reisxd WIth the latest code from master it is okay. It gets generated.

```
$ node . --patch
Downloading latest patches, integrations and cli...
No device found! Fallback to only build.
[revanced-cli-2.3.2-all.jar] ✓
[revanced-patches-2.10.1.dex] ✓
[revanced-patches-2.10.1.jar] ✓
[app-release-unsigned.apk] ✓
[microg.apk] ✓
? Select a YT version to download. 17.25.35
Downloading the YouTube APK, this'll take some time!
[youtube.apk] ✓
Download complete!
Building ReVanced, please be patient!
INFO: Decoding resources 
INFO: Reading dex files 
INFO: Merging ./revanced/app-release-unsigned.apk 
INFO: Skipping enable-wide-searchbar: Explicitly excluded 
INFO: Skipping enable-debugging: Explicitly excluded 
INFO: fenster-swipe-controls succeeded 
INFO: seekbar-tapping succeeded 
INFO: general-ads succeeded 
INFO: video-ads succeeded 
INFO: hide-infocard-suggestions succeeded 
INFO: custom-branding succeeded 
INFO: premium-heading succeeded 
INFO: minimized-playback succeeded 
INFO: disable-fullscreen-panels succeeded 
INFO: old-quality-layout succeeded 
INFO: hide-autoplay-button succeeded 
INFO: disable-create-button succeeded 
INFO: amoled succeeded 
INFO: hide-shorts-button succeeded 
INFO: hide-cast-button succeeded 
INFO: hide-watermark succeeded 
INFO: microg-support succeeded 
INFO: custom-playback-speed succeeded 
INFO: hdr-max-brightness succeeded 
INFO: Compiling resources 
INFO: Writing modified dex files 
INFO: Writing dex file classes.dex 
INFO: Writing dex file classes2.dex 
INFO: Writing dex file classes3.dex 
INFO: Writing dex file classes4.dex 
INFO: Writing dex file classes7.dex 
INFO: Writing dex file classes6.dex 
INFO: Writing dex file classes5.dex 
INFO: Writing dex file classes8.dex 
INFO: Writing resources... 
INFO: Aligning revanced_raw.apk 
INFO: Signing revanced_aligned.apk 
INFO: Copying revanced_signed.apk to revanced.apk 
INFO: Finished 

You now can install ReVanced and MicroG by transferring revanced/revanced.apk and revaned/microg.apk!
Press any key to exit...
```
With the latest release v1.2.5 it doesn't work. So maybe you can create a release with the latest code.

**Comment by reisxd** [comment]
Should be probably fixed in 632cebefdfc5ae5e4b9cbcd0b3f1e7c3a132183e.
